src/lib/components/01-elements/link/link.tsx

```diff
@@ -1,13 +1,47 @@
+import Icon from "../icon/icon";
```

Comment:
We need a way to integrate this with the [Next Link component](https://nextjs.org/docs/pages/api-reference/components/link). We also need to add some logic to use the Link for internal link and a regular `<a>` for external links. See this component for reference (it's a Remix site but the idea is the same).

This might be a complex change, it might make sense to create a new ticket for this.